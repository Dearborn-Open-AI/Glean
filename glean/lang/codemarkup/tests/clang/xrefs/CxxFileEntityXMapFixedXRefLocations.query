query: codemarkup.CxxFileEntityXMapFixedXRefLocations _
transform: [gensort, []]
