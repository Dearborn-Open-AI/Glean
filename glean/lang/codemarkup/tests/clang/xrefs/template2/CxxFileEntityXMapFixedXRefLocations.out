[
  "@generated",
  {
    "key": {
      "trace": {
        "key": {
          "xmap": {
            "key": {
              "file": { "key": "test.cpp" },
              "fixed": [
                {
                  "target": {
                    "declaration": {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "A" },
                              "scope": { "global_": { } }
                            }
                          },
                          "kind": { "struct_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 9,
                            "columnBegin": 22,
                            "lineEnd": 11,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  },
                  "ranges": [
                    { "offset": 325, "length": 1 },
                    { "offset": 38, "length": 1 }
                  ]
                },
                {
                  "target": {
                    "declaration": {
                      "function_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": { "name": { "key": "f" } } },
                              "scope": {
                                "recordWithAccess": {
                                  "record": {
                                    "key": {
                                      "name": { "key": "A" },
                                      "scope": { "global_": { } }
                                    }
                                  },
                                  "access": 0
                                }
                              }
                            }
                          },
                          "signature": {
                            "key": {
                              "returns": { "key": "void" },
                              "parameters": [
                                {
                                  "name": { "key": "" },
                                  "type": { "key": "U" }
                                }
                              ]
                            }
                          },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 10,
                            "columnBegin": 24,
                            "lineEnd": 10,
                            "columnEnd": 39
                          }
                        }
                      }
                    }
                  },
                  "ranges": [ { "offset": 371, "length": 1 } ]
                }
              ],
              "variable": [ ],
              "bound": [
                {
                  "target": {
                    "declaration": {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "A" },
                              "scope": { "global_": { } }
                            }
                          },
                          "kind": { "struct_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 9,
                            "columnBegin": 22,
                            "lineEnd": 11,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  },
                  "uses": {
                    "key": {
                      "spans": [ { "length": 1, "offsets": [ 325, 38 ] } ],
                      "expansions": [ ],
                      "spellings": [ ]
                    }
                  }
                },
                {
                  "target": {
                    "declaration": {
                      "function_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": { "name": { "key": "f" } } },
                              "scope": {
                                "recordWithAccess": {
                                  "record": {
                                    "key": {
                                      "name": { "key": "A" },
                                      "scope": { "global_": { } }
                                    }
                                  },
                                  "access": 0
                                }
                              }
                            }
                          },
                          "signature": {
                            "key": {
                              "returns": { "key": "void" },
                              "parameters": [
                                {
                                  "name": { "key": "" },
                                  "type": { "key": "U" }
                                }
                              ]
                            }
                          },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 10,
                            "columnBegin": 24,
                            "lineEnd": 10,
                            "columnEnd": 39
                          }
                        }
                      }
                    }
                  },
                  "uses": {
                    "key": {
                      "spans": [ { "length": 1, "offsets": [ 371 ] } ],
                      "expansions": [ ],
                      "spellings": [ ]
                    }
                  }
                }
              ],
              "free": [ ]
            }
          },
          "externals": [ ],
          "targets": [ ]
        }
      },
      "xref": {
        "target": {
          "name": "A",
          "file": { "key": "test.cpp" },
          "location": {
            "range": {
              "file": { "key": "test.cpp" },
              "lineBegin": 9,
              "columnBegin": 22,
              "lineEnd": 11,
              "columnEnd": 1
            }
          },
          "destination": {
            "file": { "key": "test.cpp" },
            "span": { "start": 249, "length": 1 }
          }
        },
        "source": { "span": { "start": 325, "length": 1 } }
      },
      "entity": {
        "defn": {
          "record_": {
            "key": {
              "declaration": {
                "key": {
                  "name": {
                    "key": {
                      "name": { "key": "A" },
                      "scope": { "global_": { } }
                    }
                  },
                  "kind": { "struct_": { } },
                  "source": {
                    "file": { "key": "test.cpp" },
                    "lineBegin": 9,
                    "columnBegin": 22,
                    "lineEnd": 11,
                    "columnEnd": 1
                  }
                }
              },
              "bases": [ ],
              "members": {
                "key": [
                  {
                    "function_": {
                      "key": {
                        "name": {
                          "key": {
                            "name": { "key": { "name": { "key": "f" } } },
                            "scope": {
                              "recordWithAccess": {
                                "record": {
                                  "key": {
                                    "name": { "key": "A" },
                                    "scope": { "global_": { } }
                                  }
                                },
                                "access": 0
                              }
                            }
                          }
                        },
                        "signature": {
                          "key": {
                            "returns": { "key": "void" },
                            "parameters": [
                              { "name": { "key": "" }, "type": { "key": "U" } }
                            ]
                          }
                        },
                        "source": {
                          "file": { "key": "test.cpp" },
                          "lineBegin": 10,
                          "columnBegin": 24,
                          "lineEnd": 10,
                          "columnEnd": 39
                        }
                      }
                    }
                  }
                ]
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "trace": {
        "key": {
          "xmap": {
            "key": {
              "file": { "key": "test.cpp" },
              "fixed": [
                {
                  "target": {
                    "declaration": {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "A" },
                              "scope": { "global_": { } }
                            }
                          },
                          "kind": { "struct_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 9,
                            "columnBegin": 22,
                            "lineEnd": 11,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  },
                  "ranges": [
                    { "offset": 325, "length": 1 },
                    { "offset": 38, "length": 1 }
                  ]
                },
                {
                  "target": {
                    "declaration": {
                      "function_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": { "name": { "key": "f" } } },
                              "scope": {
                                "recordWithAccess": {
                                  "record": {
                                    "key": {
                                      "name": { "key": "A" },
                                      "scope": { "global_": { } }
                                    }
                                  },
                                  "access": 0
                                }
                              }
                            }
                          },
                          "signature": {
                            "key": {
                              "returns": { "key": "void" },
                              "parameters": [
                                {
                                  "name": { "key": "" },
                                  "type": { "key": "U" }
                                }
                              ]
                            }
                          },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 10,
                            "columnBegin": 24,
                            "lineEnd": 10,
                            "columnEnd": 39
                          }
                        }
                      }
                    }
                  },
                  "ranges": [ { "offset": 371, "length": 1 } ]
                }
              ],
              "variable": [ ],
              "bound": [
                {
                  "target": {
                    "declaration": {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "A" },
                              "scope": { "global_": { } }
                            }
                          },
                          "kind": { "struct_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 9,
                            "columnBegin": 22,
                            "lineEnd": 11,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  },
                  "uses": {
                    "key": {
                      "spans": [ { "length": 1, "offsets": [ 325, 38 ] } ],
                      "expansions": [ ],
                      "spellings": [ ]
                    }
                  }
                },
                {
                  "target": {
                    "declaration": {
                      "function_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": { "name": { "key": "f" } } },
                              "scope": {
                                "recordWithAccess": {
                                  "record": {
                                    "key": {
                                      "name": { "key": "A" },
                                      "scope": { "global_": { } }
                                    }
                                  },
                                  "access": 0
                                }
                              }
                            }
                          },
                          "signature": {
                            "key": {
                              "returns": { "key": "void" },
                              "parameters": [
                                {
                                  "name": { "key": "" },
                                  "type": { "key": "U" }
                                }
                              ]
                            }
                          },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 10,
                            "columnBegin": 24,
                            "lineEnd": 10,
                            "columnEnd": 39
                          }
                        }
                      }
                    }
                  },
                  "uses": {
                    "key": {
                      "spans": [ { "length": 1, "offsets": [ 371 ] } ],
                      "expansions": [ ],
                      "spellings": [ ]
                    }
                  }
                }
              ],
              "free": [ ]
            }
          },
          "externals": [ ],
          "targets": [ ]
        }
      },
      "xref": {
        "target": {
          "name": "A",
          "file": { "key": "test.cpp" },
          "location": {
            "range": {
              "file": { "key": "test.cpp" },
              "lineBegin": 9,
              "columnBegin": 22,
              "lineEnd": 11,
              "columnEnd": 1
            }
          },
          "destination": {
            "file": { "key": "test.cpp" },
            "span": { "start": 249, "length": 1 }
          }
        },
        "source": { "span": { "start": 363, "length": 1 } }
      },
      "entity": {
        "defn": {
          "record_": {
            "key": {
              "declaration": {
                "key": {
                  "name": {
                    "key": {
                      "name": { "key": "A" },
                      "scope": { "global_": { } }
                    }
                  },
                  "kind": { "struct_": { } },
                  "source": {
                    "file": { "key": "test.cpp" },
                    "lineBegin": 9,
                    "columnBegin": 22,
                    "lineEnd": 11,
                    "columnEnd": 1
                  }
                }
              },
              "bases": [ ],
              "members": {
                "key": [
                  {
                    "function_": {
                      "key": {
                        "name": {
                          "key": {
                            "name": { "key": { "name": { "key": "f" } } },
                            "scope": {
                              "recordWithAccess": {
                                "record": {
                                  "key": {
                                    "name": { "key": "A" },
                                    "scope": { "global_": { } }
                                  }
                                },
                                "access": 0
                              }
                            }
                          }
                        },
                        "signature": {
                          "key": {
                            "returns": { "key": "void" },
                            "parameters": [
                              { "name": { "key": "" }, "type": { "key": "U" } }
                            ]
                          }
                        },
                        "source": {
                          "file": { "key": "test.cpp" },
                          "lineBegin": 10,
                          "columnBegin": 24,
                          "lineEnd": 10,
                          "columnEnd": 39
                        }
                      }
                    }
                  }
                ]
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "trace": {
        "key": {
          "xmap": {
            "key": {
              "file": { "key": "test.cpp" },
              "fixed": [
                {
                  "target": {
                    "declaration": {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "A" },
                              "scope": { "global_": { } }
                            }
                          },
                          "kind": { "struct_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 9,
                            "columnBegin": 22,
                            "lineEnd": 11,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  },
                  "ranges": [
                    { "offset": 325, "length": 1 },
                    { "offset": 38, "length": 1 }
                  ]
                },
                {
                  "target": {
                    "declaration": {
                      "function_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": { "name": { "key": "f" } } },
                              "scope": {
                                "recordWithAccess": {
                                  "record": {
                                    "key": {
                                      "name": { "key": "A" },
                                      "scope": { "global_": { } }
                                    }
                                  },
                                  "access": 0
                                }
                              }
                            }
                          },
                          "signature": {
                            "key": {
                              "returns": { "key": "void" },
                              "parameters": [
                                {
                                  "name": { "key": "" },
                                  "type": { "key": "U" }
                                }
                              ]
                            }
                          },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 10,
                            "columnBegin": 24,
                            "lineEnd": 10,
                            "columnEnd": 39
                          }
                        }
                      }
                    }
                  },
                  "ranges": [ { "offset": 371, "length": 1 } ]
                }
              ],
              "variable": [ ],
              "bound": [
                {
                  "target": {
                    "declaration": {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "A" },
                              "scope": { "global_": { } }
                            }
                          },
                          "kind": { "struct_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 9,
                            "columnBegin": 22,
                            "lineEnd": 11,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  },
                  "uses": {
                    "key": {
                      "spans": [ { "length": 1, "offsets": [ 325, 38 ] } ],
                      "expansions": [ ],
                      "spellings": [ ]
                    }
                  }
                },
                {
                  "target": {
                    "declaration": {
                      "function_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": { "name": { "key": "f" } } },
                              "scope": {
                                "recordWithAccess": {
                                  "record": {
                                    "key": {
                                      "name": { "key": "A" },
                                      "scope": { "global_": { } }
                                    }
                                  },
                                  "access": 0
                                }
                              }
                            }
                          },
                          "signature": {
                            "key": {
                              "returns": { "key": "void" },
                              "parameters": [
                                {
                                  "name": { "key": "" },
                                  "type": { "key": "U" }
                                }
                              ]
                            }
                          },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 10,
                            "columnBegin": 24,
                            "lineEnd": 10,
                            "columnEnd": 39
                          }
                        }
                      }
                    }
                  },
                  "uses": {
                    "key": {
                      "spans": [ { "length": 1, "offsets": [ 371 ] } ],
                      "expansions": [ ],
                      "spellings": [ ]
                    }
                  }
                }
              ],
              "free": [ ]
            }
          },
          "externals": [ ],
          "targets": [ ]
        }
      },
      "xref": {
        "target": {
          "name": "f",
          "file": { "key": "test.cpp" },
          "location": {
            "range": {
              "file": { "key": "test.cpp" },
              "lineBegin": 10,
              "columnBegin": 24,
              "lineEnd": 10,
              "columnEnd": 39
            }
          },
          "destination": {
            "file": { "key": "test.cpp" },
            "span": { "start": 288, "length": 1 }
          }
        },
        "source": { "span": { "start": 371, "length": 1 } }
      },
      "entity": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "f" } } },
                  "scope": {
                    "recordWithAccess": {
                      "record": {
                        "key": {
                          "name": { "key": "A" },
                          "scope": { "global_": { } }
                        }
                      },
                      "access": 0
                    }
                  }
                }
              },
              "signature": {
                "key": {
                  "returns": { "key": "void" },
                  "parameters": [
                    { "name": { "key": "" }, "type": { "key": "U" } }
                  ]
                }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 10,
                "columnBegin": 24,
                "lineEnd": 10,
                "columnEnd": 39
              }
            }
          }
        }
      }
    }
  }
]